test/view/study/study_list_screen_test.dart

```diff
@@ -0,0 +1,694 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:http/testing.dart';
+import 'package:lichess_mobile/src/network/http.dart';
+import 'package:lichess_mobile/src/view/study/study_list_screen.dart';
+
+import '../../test_helpers.dart';
+import '../../test_provider_scope.dart';
+
+void main() {
+  group('Study list screen', () {
+    final mockClient = MockClient((request) {
+      if (request.url.path == '/study/all/hot') {
+        if (request.url.queryParameters['page'] == '1') {
+          return mockResponse(kStudyAllHotPage1Response, 200);
+        }
+        if (request.url.queryParameters['page'] == '2') {
+          return mockResponse(kStudyAllHotPage2Response, 200);
+        }
+        return mockResponse(
+          emptyPage(int.parse(request.url.queryParameters['page']!)),
+          200,
+        );
+      }
+      if (request.url.path == '/study/search') {
+        if (request.url.queryParameters['q'] == 'Magnus') {
+          return mockResponse(
+            '''
+{
+  "paginator": {
+    "currentPage": 1,
+    "maxPerPage": 16,
+    "currentPageResults": [
+      {
+        "id": "g26XbGpT",
+        "name": "Magnus Carlsen Games",
+        "liked": false,
+        "likes": 1,
+        "updatedAt": 1723817543350,
+        "owner": {
+          "name": "tom-anders",
+          "id": "tom-anders"
+        },
+        "chapters": [
+          "Chapter 1",
+          "Chapter 2"
+        ],
+        "topics": [ ],
+        "members": [ ]
+      }
+    ],
+    "previousPage": null,
+    "nextPage": null,
+    "nbResults": 1,
+    "nbPages": 1
+  }
+}
+              ''',
+            200,
+          );
+        }
+      }
+      return mockResponse('', 404);
+    });
+
+    testWidgets('Scrolling down loads next page', (WidgetTester tester) async {
+      final app = await makeTestProviderScopeApp(
+        tester,
+        home: const StudyListScreen(),
+        overrides: [
+          lichessClientProvider.overrideWith((ref) {
+            return LichessClient(mockClient, ref);
+          }),
+        ],
+      );
+      await tester.pumpWidget(app);
+      await tester.pumpAndSettle();
+
+      expect(find.text('First Study Page 1'), findsOneWidget);
+      expect(find.text('First Study Page 2'), findsNothing); // On page 2
+
+      // SearchBar also implements Scrollable, so we need to explicitly specify the ListView's Scrollable here
+      await tester.scrollUntilVisible(
+        find.text('First Study Page 2'),
+        200,
+        scrollable: find.descendant(
+          of: find.byType(ListView),
+          matching: find.byType(Scrollable),
+        ),
+      );
+
+      await tester.pumpAndSettle();
```

Comment:
Also explain why there is a `pumpAndSettle` at the end.

Would be good to track the requests made and add a check to confirm that the requests match the expected.

Reply:
Ah, this `pumpAndSettle()` was needed because

```
WidgetsBinding.instance.addPostFrameCallback((_) {
   setState(() {
      requestedNextPage = false;
   });
});
```
causes a crash otherwise. But, thanks to the helpful flutter error message, we can add an `if (mounted)` check to fix this instead.

Reply:
> Would be good to track the requests made and add a check to confirm that the requests match the expected.

Added helper variables to track the requests, not sure if this is the most elegant solution though